grakn-engine/src/main/java/ai/grakn/engine/bootup/BootupProcessExecutor.java

```diff
@@ -140,24 +144,30 @@ public void stopProcessIfRunning(Path pidFile, String programName) {
 
     public void processStatus(Path storagePid, String name) {
         if (isProcessRunning(storagePid)) {
-            System.out.println(name+": RUNNING");
+            System.out.println(name + ": RUNNING");
         } else {
-            System.out.println(name+": NOT RUNNING");
+            System.out.println(name + ": NOT RUNNING");
         }
     }
 
     private void stopProcess(Path pidFile) {
-        int pid = retrievePid(pidFile);
-        if (pid < 0 ) return;
+        String pid = retrievePid(pidFile);
+        if (pid == null) return;
         kill(pid);
-        waitUntilStopped(pidFile, pid);
+        waitUntilStopped(pidFile);
     }
 
-    private void kill(int pid) {
-        executeAndWait(Arrays.asList(SH, "-c", "kill " + pid), null);
+    private List<String> killProcessCommand(String pid){
+        if (isWindows()) {
+            return Arrays.asList("cmd", "/c", "taskkill /F /PID "+ pid.trim());
+        } else {
+            return Arrays.asList(SH, "-c", "kill " + pid.trim());
+        }
     }
 
-    private BootupProcessResult kill(int pid, String signal) {
-        return executeAndWait(Arrays.asList(SH, "-c", "kill -" + signal + " " + pid), null);
+    private void kill(String pid) { executeAndWait(killProcessCommand(pid), null); }
+
+    private boolean isWindows() {
+        return System.getProperty("os.name").toLowerCase().contains("win");
```

Comment:
do we know what will happen if run under cygwin?

Reply:
nope, we should test it at some point. I took this check directly from another library - I will check if there is a more thorough and correct check

Reply:
apparently it yields `posix` https://stackoverflow.com/questions/32612235/what-is-os-name-on-cygwin